src/JWT.Extensions.AspNetCore/JwtAuthenticationEvents.cs

```diff
@@ -0,0 +1,99 @@
+namespace JWT.Extensions.AspNetCore;
+
+using System;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+
+public class JwtAuthenticationEvents
+{
+    private static readonly Action<ILogger, Exception> _logMissingHeader;
+    private static readonly Action<ILogger, string, string, Exception> _logIncorrectScheme;
+    private static readonly Action<ILogger, Exception> _logEmptyHeader;
+    private static readonly Action<ILogger, Exception> _logSuccessfulTicket;
+    private static readonly Action<ILogger, string, Exception> _logFailedTicket;
+
+    static JwtAuthenticationEvents()
+    {
+        _logMissingHeader = LoggerMessage.Define(
+            LogLevel.Information,
+            10,
+            $"Header {nameof(HeaderNames.Authorization)} is empty, returning none");
+
+        _logIncorrectScheme = LoggerMessage.Define<string, string>(
+            LogLevel.Information,
+            11,
+            $"Header {nameof(HeaderNames.Authorization)} scheme is {{0}}, expected {{1}}, returning none");
```

Comment:
The official recommendation is to use named placeholders, not numbered. Or this is the already existing (my) code? Should we change it?

Reply:
I think I just copied what was already there. But happy to upgrade it. :)